src/backend/distributed/executor/executor.c

```diff
@@ -549,10 +559,36 @@ StartDistributedExecution(DistributedExecution *execution)
 
 			if (TaskListRequires2PC(taskList))
 			{
+				/*
+				 * Although using two phase commit protocol is an independent decision than
+				 * failing on any error, we prefer to couple them. Our motivation is that
+				 * the failures are rare, and we prefer to avoid marking placements invalid
+				 * in case of failures.
+				 */
 				CoordinatedTransactionUse2PC();
+
```

Comment:
Kind of wondering whether we actually still need this line. Would `Activate2PCIfModifyingTransactionExpandsToNewNode` be sufficient?

Reply:
Agreed